Update FilmicSharpen, qUINT, and MXAO

BrimsonBhin opened the pull request:
Ported changes from original authors:
- **FilmicSharpen:** Some GUI enhancements regarding REC 601 & 709, comments added for easier updates
- **MXAO:** Tweaks from mainstream Reshade repo, formatting fixes from your port
- Watchdogs.fx format fix
- Lightroom updates from Marty

**Note:** The authors of qUINT and FilmicSharpen have not updated their shaders for months. Therefore, these changes are here to stay for some time.

Commit:
Update FilmicSharpen.fx

Commit:
Update WatchDogs.fx

Commit:
update qUINT dependency

Commit:
Revert messed up gshade support

Commit:
Update qUINT_dof.fx

Commit:
Update qUINT_lightroom.fx

Commit:
Revert messup

Commit:
Update qUINT_ssr.fx

Commit:
Update qUINT_ssr.fx

Commit:
Update qUINT_ssr.fx

Commit:
Update MXAO 4.0.2 EX.fx

Commit:
Update qUINT_lightroom.fx

Mortalitas commented on Shaders/MXAO 4.0.2 EX.fx:
```diff
@@ -24,75 +24,76 @@
 #endif
 
 #ifndef MXAO_SMOOTHNORMALS
- #define MXAO_SMOOTHNORMALS             1       //[0 or 1]      This feature makes low poly surfaces smoother, especially useful on older games.
```
This is intentionally set to 1 in GShade, as the primary concern of the build is Final Fantasy XIV support.

Mortalitas commented on Shaders/MXAO 4.0.2 EX.fx:
```diff
@@ -24,75 +24,76 @@
 #endif
 
 #ifndef MXAO_TWO_LAYER
- #define MXAO_TWO_LAYER                 1       //[0 or 1]      Splits MXAO into two separate layers that allow for both large and fine AO.
```
This is intentionally set to 1 in GShade, as the primary concern of the build is Final Fantasy XIV support.

Mortalitas commented on Shaders/MXAO 4.0.2 EX.fx:
```diff
@@ -24,75 +24,76 @@
-	ui_type = "combo";
+        ui_type = "combo";
         ui_label = "Sample Quality";
-	ui_items = "Very Low (4)\0Low (8)\0Medium (16)\0High (24)\0Very High (32)\0Ultra (64)\0Maximum (255)\0";
```
Values were added here for the sake of end-user clarity.

Mortalitas commented on Shaders/MXAO 4.0.2 EX.fx:
```diff
@@ -24,75 +24,76 @@
 uniform float MXAO_SAMPLE_RADIUS <
-	ui_type = "slider";
-	ui_min = 0.5; ui_max = 20.0;
+        ui-ui_type = "slider"
```
This isn't terminated and I'm not familiar with a ui-ui-type.

Mortalitas commented on Shaders/MXAO 4.0.2 EX.fx:
```diff
@@ -24,75 +24,76 @@
 uniform float MXAO_SAMPLE_NORMAL_BIAS <
-	ui_type = "slider";
-	ui_min = 0.0; ui_max = 2.0;
+        ui-ui_type = "slider"
```
This isn't terminated and I'm not familiar with a ui-ui-type.

Mortalitas commented on Shaders/MXAO 4.0.2 EX.fx:
```diff
@@ -24,75 +24,76 @@
 
 uniform float MXAO_GLOBAL_RENDER_SCALE <
-	ui_type = "slider";
+        ui-ui_type = "slider"
```
This isn't terminated and I'm not familiar with a ui-ui-type.

Mortalitas commented on Shaders/MXAO 4.0.2 EX.fx:
```diff
@@ -24,75 +24,76 @@
 uniform float MXAO_SSAO_AMOUNT <
-	ui_type = "slider";
-	ui_min = 0.00; ui_max = 6.00;
+        ui-ui_type = "slider"
```
This isn't terminated and I'm not familiar with a ui-ui-type.

Mortalitas commented on Shaders/MXAO 4.0.2 EX.fx:
```diff
@@ -24,75 +24,76 @@
         uniform float MXAO_SSIL_AMOUNT <
-        	ui_type = "slider";
-        	ui_min = 0.00; ui_max = 12.00;
+                ui-ui_type = "slider"
```
This isn't terminated and I'm not familiar with a ui-ui-type.

Mortalitas commented on Shaders/MXAO 4.0.2 EX.fx:
```diff
@@ -24,75 +24,76 @@
         uniform float MXAO_SSIL_SATURATION <
-        	ui_type = "slider";
-        	ui_min = 0.00; ui_max = 3.00;
+                ui-ui_type = "slider"
```
This isn't terminated and I'm not familiar with a ui-ui-type.

Mortalitas commented on Shaders/MXAO 4.0.2 EX.fx:
```diff
@@ -24,75 +24,76 @@
+        
         uniform float MXAO_SSIL_SATURATION_FILTER <
-				ui_type = "slider";
+                ui-ui_type = "slider"
```
This isn't terminated and I'm not familiar with a ui-ui-type.

Mortalitas commented on Shaders/MXAO 4.0.2 EX.fx:
```diff
@@ -24,75 +24,76 @@
         
         uniform float MXAO_SSIL_GAMMA <
-				ui_type = "slider";
+                ui-ui_type = "slider"
```
This isn't terminated and I'm not familiar with a ui-ui-type.

Mortalitas commented on Shaders/MXAO 4.0.2 EX.fx:
```diff
@@ -101,60 +102,60 @@ uniform float MXAO_SSAO_AMOUNT <
-           	ui_min = 0.1; ui_max = 2.00;
-                   ui_label = "Fine AO Scale";
-           	ui_tooltip = "Multiplier of Sample Radius for fine geometry. A setting of 0.5 scans the geometry at half the radius of the main AO.";
+                ui-ui_type = "slider"
```
This isn't terminated and I'm not familiar with a ui-ui-type.

Mortalitas commented on Shaders/MXAO 4.0.2 EX.fx:
```diff
@@ -101,60 +102,60 @@ uniform float MXAO_SSAO_AMOUNT <
-             ui_min = 0.00; ui_max = 2.00;
-             ui_label = "Fine AO intensity multiplier";
-             ui_tooltip = "Intensity of small scale AO / IL.";
+                ui-ui_type = "slider"
```
This isn't terminated and I'm not familiar with a ui-ui-type.

Mortalitas commented on Shaders/MXAO 4.0.2 EX.fx:
```diff
@@ -101,60 +102,60 @@ uniform float MXAO_SSAO_AMOUNT <
-             ui_min = 0.00; ui_max = 2.00;
-             ui_label = "Coarse AO intensity multiplier";
-             ui_tooltip = "Intensity of large scale AO / IL.";
+                ui-ui_type = "slider"
```
This isn't terminated and I'm not familiar with a ui-ui-type.

Mortalitas commented on Shaders/MXAO 4.0.2 EX.fx:
```diff
@@ -101,60 +102,60 @@ uniform float MXAO_SSAO_AMOUNT <
 
 uniform float MXAO_GAMMA <
-		ui_type = "slider";
+        ui-ui_type = "slider"
```
This isn't terminated and I'm not familiar with a ui-ui-type.

Mortalitas commented on Shaders/MXAO 4.0.2 EX.fx:
```diff
@@ -506,7 +507,7 @@ void PS_AmbientObscurance(in MXAO_VSOUT MXAO, out float4 color : SV_Target0)
                         if(fAO > 0.1)
                         {
-        			float3 fIL = tex2Dlod(sMXAO_ColorTex, float4(sampleUV,0,sampleMIP + MXAO_MIPLEVEL_IL)).xyz;
+        			const float3 fIL = tex2Dlod(sMXAO_ColorTex, float4(sampleUV,0,sampleMIP + MXAO_MIPLEVEL_IL)).xyz;
```
This variable cannot be const.

Mortalitas commented on Shaders/qUINT_bloom.fx:
```diff
@@ -21,7 +21,7 @@
 =============================================================================*/
 
 #ifndef SAMPLE_HQ
- #define SAMPLE_HQ 1
```
This is intentionally set to 1 in GShade, as the primary concern of the build is Final Fantasy XIV support.

Mortalitas commented on Shaders/qUINT_dof.fx:
```diff
@@ -122,7 +122,7 @@ uniform float fADOF_ShapeRadius <
 uniform float fADOF_SmootheningAmount <
     ui_type = "slider";
     ui_min = 0.0;
-    ui_max = 200.0;
```
This was intentionally set as a cap as a user suggested it.

Mortalitas committed:
Rename qUINT_qmxao.fx to qUINT_mxao.fx

Not sure why this was renamed? For the sake of compatibility with existing presets, filenames need to be static.

Mortalitas reviewed (changes requested):
Some preprocessor definitions need reverting as the changes were intentional, and I think there was a find and replace issue?

Mortalitas committed:
Fix ui-types, termination, preprocessor definitions

And one stray const variable.

Mortalitas committed:
Reverting some stuff.

Mortalitas committed:
Reverting a preprocessor definition

Mortalitas committed:
Reverting a variable cap

Mortalitas committed:
Missed a variable cap

ghost commented:
Ah yes that was a find and replace issue. Thanks for that
